src/frequenz/sdk/actor/power_distributing/_distribution_algorithm/_distribution_algorithm.py

```diff
@@ -249,24 +327,46 @@ def _compute_battery_availability_ratio(
                 of all battery ratios in the list.
         """
         total_capacity = self._total_capacity(components)
-        battery_availability_ratio: list[tuple[InvBatPair, float, float]] = []
+        battery_availability_ratio: list[AvailabilityRatio] = []
         total_battery_availability_ratio: float = 0.0
 
         for pair in components:
-            battery, inverter = pair
+            battery, inverters = pair
             capacity_ratio = battery.capacity / total_capacity
-            soc_factor = pow(
+            soc_factor: float = pow(
                 available_soc[battery.component_id], self._distributor_exponent
             )
 
             ratio = capacity_ratio * soc_factor
-            battery_availability_ratio.append(
-                (pair, excl_bounds[inverter.component_id], ratio)
-            )
+
+            if len(inverters) > 1:
+                # If there are more than one inverter, we need to distribute
+                # the ratio between them.
+                #
+                # This is also the place where different strategies
+                # to handle multiple inverters can be implemented.
+                per_inverter_ratio = ratio / len(inverters)
+                for inverter in inverters:
+                    battery_availability_ratio.append(
+                        AvailabilityRatio(
+                            inverter.component_id,
+                            excl_bounds[inverter.component_id],
+                            per_inverter_ratio,
+                        )
+                    )
```

Comment:
wasn't the consensus to use each inverter to its max power, before using the next one?  I think people on the ticket said otherwise, we lose power to inverter inefficiencies and self-consumptions.  

Also, it maybe easier to consider it a single device for the first pass,  so that you don't have to divide the exlusion bounds (if any) into several inverters, or try to give minimum power for all inverters.  

And after the stage as a single target power is done, you can just redistribute it to the inverters, filling up each one before going to the next.

Reply:
Yes, but after talking with a few more ppl we agreed that for a first version we take the most naive approach

Reply:
Have you tried doing some tests with exclusion bounds?  I still feel this is the way to go:

> Also, it maybe easier to consider it a single device for the first pass, so that you don't have to divide the exlusion bounds (if any) into several inverters, or try to give minimum power for all inverters.
> 
> And after the stage as a single target power is done, you can just redistribute it to the inverters, filling up each one before going to the next.

Reply:
Yeah, I am right now working on that exclusion part, after I added some tests I noticed that that part needs some work.

It is a tougher problem with those, indeed